plugin/clone/src/clone_common.cc

```diff
@@ -98,16 +81,118 @@ int Ha_clone_common_cbk::synchronize_engines() {
   // Size reverse-engineered from ER_CLONE_SERVER_TRACE used by log_error. No
   // way to track it automatically, but very unlikely it would silently shrink.
   char msg_buf[512];
-  snprintf(msg_buf, sizeof(msg_buf), "engine positions: %.*s",
+  snprintf(msg_buf, sizeof(msg_buf), "%.*s: %.*s",
+           static_cast<int>(object_name.size()), object_name.data(),
            static_cast<int>(json_str_buf.length()), json_str_buf.ptr());
   log_error(nullptr, false, 0, msg_buf);
 
   assert(json_dom->json_type() == enum_json_type::J_OBJECT);
+  return *static_cast<const Json_object *>(json_dom);
+}
+
+int Ha_clone_common_cbk::populate_synchronization_coordinates(
+    const Json_object &local_repl_info,
+    Key_Values &synchronization_coordinates) {
+  // get synchronization coordinates from log status
+  static const std::string gtid_executed_key_str = "gtid_executed";
+  const auto &gtid_executed_json = local_repl_info.get(gtid_executed_key_str);
+  assert(gtid_executed_json->json_type() == enum_json_type::J_STRING);
+  const auto &gtid_executed_str_json =
+      static_cast<const Json_string &>(*gtid_executed_json);
+  const auto &gtid_executed_str = gtid_executed_str_json.value();
+  synchronization_coordinates = {{"gtid_from_log_status", gtid_executed_str}};
+
+  static const std::string binary_log_file_key_str = "binary_log_file";
+  const auto &binary_log_file_json_ptr =
+      local_repl_info.get(binary_log_file_key_str);
+  if (binary_log_file_json_ptr == nullptr) {
+    // if there is no binlog file, don't populate
+    return 0;
+  }
+  const auto &binary_log_file_json = *binary_log_file_json_ptr;
+  assert(binary_log_file_json.json_type() == enum_json_type::J_STRING);
+  const auto &binary_log_file_str_json =
+      static_cast<const Json_string &>(binary_log_file_json);
+  const auto &binary_log_file_str = binary_log_file_str_json.value();
+  synchronization_coordinates.push_back(
+      {binary_log_file_key_str, binary_log_file_str});
+
+  static const std::string binary_log_position_key_str = "binary_log_position";
+  const auto &binary_log_position_json_ptr =
+      local_repl_info.get(binary_log_position_key_str);
+  if (binary_log_position_json_ptr == nullptr) {
+    // if there is no binlog offset, don't populate
+    return 0;
+  }
+  const auto &binary_log_position_json = *binary_log_position_json_ptr;
+  assert(binary_log_position_json.json_type() == enum_json_type::J_INT);
+  const auto &binary_log_position_int_json =
+      static_cast<const Json_int &>(binary_log_position_json);
+  const auto &binary_log_position_int = binary_log_position_int_json.value();
+  synchronization_coordinates.push_back(
+      {binary_log_position_key_str, std::to_string(binary_log_position_int)});
+
+  // get gtid from binlog file/pos
```

Comment:
On the second thought, please add a comment here why we are getting the same information second time in a different way (hunting for a bug, will update once debugged)